server/lib/chat/chat-service.test.ts

```diff
@@ -0,0 +1,1505 @@
+    const userSocketsManager = new UserSocketsManager(nydus, sessionLookup, async () => {})
+    const publisher = new TypedPublisher(nydus)
+
+    chatService = new ChatService(publisher, userSocketsManager)
+    connector = new NydusConnector(nydus, sessionLookup)
+
+    client1 = connector.connectClient(user1, USER1_CLIENT_ID)
+    client2 = connector.connectClient(user2, USER2_CLIENT_ID)
+
+    jest.clearAllMocks()
+    clearTestLogs(nydus)
+  })
+
+  describe('joinInitialChannel', () => {
+    const addUserToChannelMock = asMockedFunction(addUserToChannel)
+    const addMessageToChannelMock = asMockedFunction(addMessageToChannel)
+
+    beforeEach(async () => {
+      // NOTE(2Pac): We pre-join user2 to the channel so we can check it sees correct websocket
+      // messages from user1 joining initially.
+      await joinUserToChannel(
+        user2,
+        shieldBatteryChannel,
+        user2ShieldBatteryChannelEntry,
+        joinUser2ShieldBatteryChannelMessage,
+      )
+
+      asMockedFunction(findUserById).mockResolvedValue(user1)
+      asMockedFunction(getChannelInfo).mockResolvedValue([shieldBatteryChannel])
+      addUserToChannelMock.mockResolvedValue(user1ShieldBatteryChannelEntry)
+      addMessageToChannelMock.mockResolvedValue(joinUser1ShieldBatteryChannelMessage)
+    })
+
+    test("should throw if user doesn't exist", async () => {
+      asMockedFunction(findUserById).mockResolvedValue(undefined)
+
+      await expect(
+        chatService.joinInitialChannel(user1.id, dbClient, Promise.resolve()),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(`"User doesn't exist"`)
+    })
+
+    test('works when user exists', async () => {
+      await chatService.joinInitialChannel(user1.id, dbClient, Promise.resolve())
+
+      expect(addUserToChannelMock).toHaveBeenCalledWith(user1.id, shieldBatteryChannel.id, dbClient)
+      expect(addMessageToChannelMock).toHaveBeenCalledWith(
+        user1.id,
+        shieldBatteryChannel.id,
+        { type: ServerChatMessageType.JoinChannel },
+        dbClient,
+      )
+      expect(client2.publish).toHaveBeenCalledWith(getChannelPath(shieldBatteryChannel.id), {
+        action: 'join2',
+        user: user1,
+        message: {
+          id: joinUser1ShieldBatteryChannelMessage.msgId,
+          type: joinUser1ShieldBatteryChannelMessage.data.type,
+          channelId: joinUser1ShieldBatteryChannelMessage.channelId,
+          userId: joinUser1ShieldBatteryChannelMessage.userId,
+          time: Number(joinUser1ShieldBatteryChannelMessage.sent),
+        },
+      })
+      expect(nydus.subscribeClient).toHaveBeenCalledWith(
+        client1,
+        getChannelPath(shieldBatteryChannel.id),
+        {
+          action: 'init3',
+          channelInfo: shieldBatteryChannel,
+          activeUserIds: [user2.id, user1.id],
+          selfPermissions: channelPermissions,
+        },
+      )
+      expect(nydus.subscribeClient).toHaveBeenCalledWith(
```

Comment:
This particular check is fine to check on subscribeClient, I think (because there isn't a great way to check this otherwise since it has no initial data)